### PR TITLE
Bench 331: Framer Motion Page Transitions

### PR DESCRIPTION
**Pull request by harrystead2**
answerkingdem9

### CODE REVIEW

**Comment by milkshakeuk** [comment]
I don't think you should be able to check out if there are no line items (the button should be disabled until a line item exists)

**Comment by harrystead2** [comment]
> I don't think you should be able to check out if there are no line items (the button should be disabled until a line item exists)

Yes true, we'll address that in Audrius' PR

File: src/pages/Home/Home.tsx [hunk 1]
```diff
@@ -19,24 +20,26 @@ export const HomePage = (): ReactElement => {
...
       </Helmet>
-      <div className="grid grid-cols-2 gap-2">
-        <div className="ml-44 mt-20 flex flex-col items-center justify-center">
-          <div className="mb-10 text-center text-[80px] text-white">
-            <h3 className="font-poly italic leading-tight">
-              The
-              <img src={Logo} alt="logo" className="ml-5 mb-4 inline-block h-12" />
-              <span className="font-poppins not-italic">nswer</span> to <br></br> your cravings
-            </h3>
+      <PageTransition>
```
Comment (bethcryer) [inline src/pages/Home/Home.tsx]: Do you think we could avoid the code repetition by making a general Page component, with a prop for the title and a prop for the contents that are children of the PageTransition component on each page?

Then the home page would become something like <Page title="Home - Answer King" content=<HomeContent /> />

File: src/pages/Menu/Menu.tsx [hunk 2]
```diff
@@ -30,20 +31,22 @@ export const MenuPage = (): ReactElement => {
...
-            )}
-          />
+      <PageTransition>
+        <MenuCategories selectedCategory={selectedCategory} setSelectedCategory={setSelectedCategory} categories={categories.data} />
+        <div className="mt-10 grid grid-cols-12">
+          <div className="col-span-1"></div>
+          <div className="col-span-8 text-center">
+            <MenuItems
+              products={products.data.filter(
+                (product) => product.retired === false && product.categories?.find((categoryId) => categoryId === selectedCategory)
```
Comment (bethcryer) [inline src/pages/Menu/Menu.tsx]: does `!product.retired` work here? it's just a boolean so it shouldn't ever be anything but true or false, right?

**Review by bethcryer — DISMISSED** [review]

File: src/pages/Menu/Menu.tsx (hunk 2 above)
Comment (milkshakeuk) [reply src/pages/Menu/Menu.tsx]: looks like harry is doing a [strict equals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) `===` , if he did `!product.retired`, that's going to be a [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) check.

Not sure if this makes a difference, I guess the former is safer though, and no [type coercion](https://developer.mozilla.org/en-US/docs/Glossary/Type_coercion) needs to take place.

**Review by bethcryer — APPROVED** [review]

**Review by AudriusPutys — APPROVED** [review]

**Pull request merged**